File: code/generate_strategies/RA(name)/java/java/function_BeautifulSoup_find_all.java
package com.example;

import org.jsoup.Jsoup;
import org.jsoup.nodes.Document;
import org.jsoup.select.Elements;
import java.util.stream.Collectors;

public class HTMLParser {
    public String extract_links(String html_content) {
        Document doc = Jsoup.parse(html_content);
        Elements links = doc.select("a[href]");

        if (links.isEmpty()) {
            return "None";
        }

        return links.stream()
                .map(link -> link.attr("href"))
                .collect(Collectors.joining(","));
    }
}

File: manual/data/python/src/function_BeautifulSoup_find_all.py
from bs4 import BeautifulSoup

class HTMLParser:
    def extract_links(self, html_content: str) -> str:
        soup = BeautifulSoup(html_content, 'html.parser')
        links = [a.get('href') for a in soup.find_all('a') if a.get('href')]
        return ",".join(links) if links else "None"
